src/LibraryViewExtensionWebView2/LibraryViewController.cs

```diff
@@ -187,6 +188,19 @@ public void CreateNode(string nodeName)
                 {
                     this.disableObserver = true;
                 }
+
+                // check to make sure a custom node cannot be added to its own workspace.
+                if (dynamoViewModel.CurrentSpace is CustomNodeWorkspaceModel customNodeWorkspaceModel)
+                {
+                    var nodeGuid = Guid.Parse(nodeName);
+
+                    if (nodeGuid.Equals(customNodeWorkspaceModel.CustomNodeId))
```

Comment:
I think @QilongTang was suggesting that in this check you iterate all the other nodes and make sure that one of them is `scopeif` OR you defer this check until they try to save the .dyf.

Reply:
if you wanted to get really fancy you could try to determine if they have the scope if hooked up correctly or at least in a state where it has even a remote chance of working...

Reply:
Got it, looking into it.

Reply:
Updated the changes to look for the scopeif node in the custom node workspace and only then allow users to place the custom node in its own workspace. Modified the notification message accordingly. 
As this freeze/hang happens mainly when that particular custom node instance is also present in the home workspace and is executed automatically, doing this check during saving the dyf wouldn't fix this case.

Reply:
what about switching them into manual mode when this occurs?

Reply:
yes made the change to set the homeworkspace to manual mode, when this case arises.